db/migrate/20170301150018_delete_orphaned_data.rb

```diff
@@ -0,0 +1,48 @@
+class DeleteOrphanedData < ActiveRecord::Migration
+  def change
+    #delete works for which there is no collection
+    Work.all.each {|w| w.destroy unless w.collection}
+    
+    #delete pages for which there are no works
+    Page.all.each {|p| p.destroy unless p.work}
+    
+    #delete document sets for which there is no collection
+    DocumentSet.all.each {|d| d.destroy unless d.collection}
+
+    #delete notes for which there is no collection
+    Note.all.each {|n| n.destroy unless n.collection}
```

Comment:
Notes are on a collection, rather than a work or page?  I wonder if this could go wrong?  

 # associations
  belongs_to :user
  belongs_to :page
  belongs_to :work
  belongs_to :collection

Any chance a note would have a work or a page, but no collection?  Any chance the collection would be wrong?